app/App/Panel/Main/Account/Requests/TransactionRequest/TxApproval/approvals/TokenSpend/index.js

```diff
@@ -35,10 +35,12 @@ function nFormat (n, digits = 2)  {
 class TokenSpend extends React.Component {
   constructor (...args) {
     super(...args)
+    const { approval: { data }, revoke } = this.props
 
-    this.decimals = this.props.approval.data.decimals || 0
-    this.requestedAmount = '0x' + new BigNumber(this.props.approval.data.amount).integerValue().toString(16)
+    this.decimals = data.decimals || 0
+    this.requestedAmount = '0x' + new BigNumber(data.amount).integerValue().toString(16)
     this.state = {
+      revoke: !!revoke,
```

Comment:
don't think we need to add this to the state

Reply:
hmmm, true, it won't be changing